### PR TITLE
 feat(switch): Convert JS to TypeScript

### PR DESCRIPTION
**Pull request by mmalerba**
depends on #4292

### CODE REVIEW

File: packages/mdc-base/component.ts [hunk 1]
```diff
@@ -114,4 +114,4 @@ class MDCComponent<FoundationType extends MDCFoundation> {
   }
 }
 
-export default MDCComponent;
+export {MDCComponent as default, MDCComponent};
```
Comment (moog16) [inline packages/mdc-base/component.ts]: just curious why you decided to export both ways.

File: packages/mdc-switch/foundation.ts [hunk 2]
```diff
@@ -21,47 +21,51 @@
...
-      addClass: (/* className: string */) => {},
-      removeClass: (/* className: string */) => {},
-      setNativeControlChecked: (/* checked: boolean */) => {},
-      setNativeControlDisabled: (/* disabled: boolean */) => {},
-    });
+  /** The default Adapter for the switch. */
+  static get defaultAdapter(): MDCSwitchAdapter {
+    return {
+      // tslint:disable:no-empty
+      addClass: () => {},
```
Comment (moog16) [inline packages/mdc-switch/foundation.ts]: I've been changing these to `() => undefined`, then removing the tslint:disables

File: packages/mdc-base/component.ts (hunk 1 above)
Comment (mmalerba) [reply packages/mdc-base/component.ts]: This way it doesn't break people using the default export, but it allows them to use the named one if they want. At some point you could deprecate the default one and eventually remove it.

Comment (moog16) [reply packages/mdc-base/component.ts]: Ok, if we do this then we should be doing this across all components then, correct?

Comment (mmalerba) [reply packages/mdc-base/component.ts]: If eliminating default exports is something you want to do eventually, then yeah I would do this kind of thing to all the ones that have default exports

File: packages/mdc-form-field/adapter.ts [hunk 3]
```diff
@@ -21,39 +21,15 @@
...
-  deactivateInputRipple() {}
+interface MDCFormFieldAdapter {
+  activateInputRipple(): void;
+  deactivateInputRipple(): void;
+  deregisterInteractionHandler<K extends keyof GlobalEventHandlersEventMap>(
+    type: K, handler: (evt: GlobalEventHandlersEventMap[K],
+  ) => void): void;
+  registerInteractionHandler<K extends keyof GlobalEventHandlersEventMap>(
+    type: K, handler: (evt: GlobalEventHandlersEventMap[K],
+  ) => void): void;
```
Comment (moog16) [inline packages/mdc-form-field/adapter.ts]: update the export default line

**Comment by moog16** [comment]
LGTM..but i think you are waiting for the form-field/selection-control PR to merge first, correct?

**Comment by mmalerba** [comment]
rebased now that dependencies are merged

File: test/unit/mdc-switch/mdc-switch.test.js [hunk 4]
```diff
@@ -27,7 +27,7 @@ import td from 'testdouble';
 
 import {supportsCssVariables} from '../../../packages/mdc-ripple/util';
 import {install as installClock} from '../helpers/clock';
-import {MDCSwitchFoundation, MDCSwitch} from '../../../packages/mdc-switch/index';
+import {MDCSwitchFoundation, MDCSwitch} from '../../../packages/mdc-switch';
```
[the remark this answers is not in the document]
Comment (mmalerba) [reply test/unit/mdc-switch/mdc-switch.test.js]: I thought I saw this in another one of these conversion PRs which is why I did it here, will undo it though

**Pull request merged**